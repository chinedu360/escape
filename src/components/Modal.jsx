// Import necessary modules and components
import PropTypes from "prop-types";
import { SimpleGrid, Center, Box } from "@chakra-ui/react";

// Define the Modal component
const Modal = ({ length, authorLen, comment }) => {
  return (
    <div className="m-2">
      {/* Create a grid with 1 column on small screens, 2 columns on medium screens, and 3 columns on large screens */}
      <SimpleGrid columns={{ base: 1, md: 2, lg: 3 }} spacing={4}>
        {/* Box for 'My Articles' */}
        <Box
          bg="white"
          height="80px"
          borderRadius="lg"
          className="border-red-300 border-[2px] rounded-md"
          boxShadow="md"
        >
          <Center p={3} color="tomato">
            {length}
          </Center>
          <Center p={1} color="gray">
            My Articles
          </Center>
        </Box>
        {/* Box for 'Our Authors' */}
        <Box
          bg="white"
          height="80px"
          borderRadius="lg"
          className="border-green-600 border-[2px] rounded-md"
        >
          <Center p={3} color="green">
            {authorLen}
          </Center>
          <Center p={1} color="gray">
            Our Authors
          </Center>
        </Box>
        {/* Box for 'Comments' */}
        <Box
          bg="white"
          height="80px"
          borderRadius="lg"
          className="border-teal-500 border-[2px] rounded-md"
        >
          <Center p={3} color="teal">
            {comment}
          </Center>
          <Center p={1} color="gray">
            Comments
          </Center>
        </Box>
      </SimpleGrid>
    </div>
  );
};

// Define propTypes for the Modal component to specify expected prop types
Modal.propTypes = {
  length: PropTypes.number, // Length of 'My Articles'
  authorLen: PropTypes.number, // Length of 'Our Authors'
  comment: PropTypes.number, // Number of 'Comments'
};

// Export the Modal component
export default Modal;
