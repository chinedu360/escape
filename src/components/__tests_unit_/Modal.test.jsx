import { render, screen } from "@testing-library/react";
import Modal from "../Modal";
import { MemoryRouter } from "react-router-dom"; // Use MemoryRouter for routing-related tests
import { vi, describe, it, expect } from "vitest";
import TestRenderer from "react-test-renderer";

describe("Modal.jsx", () => {
  it("test the children inside the Modal component", () => {
    const modal = TestRenderer.create(
      <MemoryRouter>
        <Modal />
      </MemoryRouter>
    ).toJSON();

    //modal containers
    expect(modal.type).toBe("div");
    expect(modal.children[1].type).toBe("div");
  });

  it("Snapshot", () => {
    const modal = TestRenderer.create(
      <MemoryRouter>
        <Modal />
      </MemoryRouter>
    ).toJSON();

    expect(modal).toMatchSnapshot();
  });
});
