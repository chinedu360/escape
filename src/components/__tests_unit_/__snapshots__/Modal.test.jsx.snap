// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal.jsx > Snapshot 1`] = `
<div
  className="m-2"
>
  <div
    className="css-1ghed9e"
  >
    <div
      className="border-red-300 border-[2px] rounded-md css-170zrko"
    >
      <div
        className="css-1bwftuo"
      />
      <div
        className="css-1p2ux4r"
      >
        My Articles
      </div>
    </div>
    <div
      className="border-green-600 border-[2px] rounded-md css-1n1mkoi"
    >
      <div
        className="css-daheo5"
      />
      <div
        className="css-1p2ux4r"
      >
        Our Authors
      </div>
    </div>
    <div
      className="border-teal-500 border-[2px] rounded-md css-1n1mkoi"
    >
      <div
        className="css-h53m4y"
      />
      <div
        className="css-1p2ux4r"
      >
        Comments
      </div>
    </div>
  </div>
</div>
`;
